util/querysets.py

```diff
@@ -0,0 +1,43 @@
+from projects.models import Project
+from allocations.models import Allocation
+from django.db.models import Max, Prefetch
+
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class CCQuerySets:
+    def allocationsQuerySet(self):
+        """Get all allocations from django DB."""
+        logger.debug("Allocations QS INIT")
+        alloc_qs = Allocation.objects.select_related("requestor", "reviewer")
+        return alloc_qs
+
+    def orderedAllocations(self):
+        """Sort allocations by date requested."""
+        output_qs = self.allocationsQuerySet().order_by("date_requested").reverse()
+        return output_qs
+
+    def prefetchAllocations(self, filtered_qs, input_qs):
+        """Prefetch allocations, for use by other querysets"""
+        pf = Prefetch("allocations", queryset=filtered_qs)
+        output_qs = input_qs.prefetch_related(pf)
+        return output_qs
+
+    def projectsQuerySet(self):
+        """Get all projects from django DB."""
+        logger.debug("Projects QS INIT")
+        output_qs = Project.objects.select_related("type", "pi", "field")
+        return output_qs
+
+    def orderedProjects(self):
+        """Get projects, sorted by latest allocation."""
+        filtered_qs = self.orderedAllocations()
```

Comment:
This line ensures that the allocations are sorted, before they're nested into the project query.

If we wanted to restrict the listed allocations to only ones matching a query, the filter could be applied here.

Reply:
Where exactly is this used? I don't see any references to it in the codebase.